credentials/credentials.go

```diff
@@ -77,23 +111,81 @@ func GetNewRelicLicenseKey(ctx context.Context, conf *config.Configuration) (str
 		return conf.LicenseKey, nil
 	}
 
+	var err error
+	var licenseKey string
+
 	secretId := getLicenseKeySecretId(conf)
+	if secretId != "" {
+		licenseKey, err = tryLicenseKeyFromSecret(ctx, secretId)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	parameterName := getLicenseKeyParameterName(conf)
+	if parameterName != "" {
+		licenseKey, err = tryLicenseKeyFromParameter(ctx, parameterName)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	envLicenseKey, found := os.LookupEnv(defaultSecretId)
+	if found {
+		return envLicenseKey, nil
+	}
+
+	if err != nil {
```

Comment:
Maybe a comment, since it's rare to defer error checks this long?
```suggestion
	// Check for errors while fetching from Secrets Manager or Parameter Store
	if err != nil {
```